padertorch/contrib/examples/source_separation/pit/data.py

```diff
@@ -8,25 +6,34 @@
 from paderbox.transform import stft
 
 
-def prepare_dataset(
-        db, dataset_name: str, batch_size, return_keys=None, prefetch=True, shuffle=True
-):
-    audio_keys = ['observation', 'speech_source']
+def prepare_dataset(db, dataset_name: str, batch_size, prefetch=True, shuffle=True):
+    """
+    Prepares the dataset for the training process (loading audio data, SFTF)
+
+    Args:
+        db: database object
+        dataset_name: name of the dataset that should be used
+        batch_size: batch size for the training
+        prefetch: should the data be prefetched
+        shuffle: should the data be shuffeled
+
+    Returns:
+        desired dataset of the database in prepared for the training
+    """
     dataset = db.get_dataset(dataset_name)
 
-    dataset = (
-        dataset
-        .map(partial(read_audio, audio_keys=audio_keys))
-        .map(partial(pre_batch_transform, return_keys=return_keys))
-    )
+    #Loading of the dataset and preparation of the data (STFT), the lambda function fixes the audio keys for read_audio
+    dataset = (dataset.map(read_audio).map(pre_batch_transform))
+
     if shuffle:
         dataset = dataset.shuffle(reshuffle=True)
+
+    #Splitting the dataset in batches and sorting the frames in the batch
```

Comment:
Better write "... and sorts examples in a batch w.r.t. their duration" or something similar.
The frames themselves are not sorted